backend/infrahub/core/schema/schema_branch.py

```diff
@@ -987,6 +989,24 @@ def validate_names(self) -> None:
                 ):
                     raise ValueError(f"{node.kind}: {rel.name} isn't allowed as a relationship name.")
 
+    def validate_python_keywords(self) -> None:
+        """Validate that attribute and relationship names don't use Python keywords."""
+        for name in self.all_names:
+            node = self.get(name=name, duplicate=False)
+
+            if node.kind in INTERNAL_SCHEMA_NODE_KINDS:
+                continue
+
+            # Check for Python keywords in attribute names
+            for attribute in node.attributes:
+                if keyword.iskeyword(attribute.name):
+                    raise ValueError(f"Python keyword '{attribute.name}' cannot be used as an attribute name on '{node.kind}'")
+
+            # Check for Python keywords in relationship names
+            for relationship in node.relationships:
+                if keyword.iskeyword(relationship.name):
+                    raise ValueError(f"Python keyword '{relationship.name}' cannot be used as a relationship name on '{node.kind}'")
```

Comment:
This change with the relationships is a breaking change as it worked before where as on the attribute side it would always cause an error, for that reason we'd need to update this check to ensure that we only require relationship names to not use Python keywords if the schema is configured in strict mode.